tron/api/controller.py

```diff
@@ -174,11 +175,19 @@ def update_config(self, name, content, config_hash):
         if self.config_manager.get_hash(name) != config_hash:
             return "Configuration has changed. Please try again."
 
+        old_config = copy(self.read_config(name)['config'])
```

Comment:
Do we need to copy, if we're reading from the file?

Reply:
Oh, yea, I was being dumb here.